pkg/appender/store.go

```diff
@@ -158,22 +161,30 @@ func (cs *chunkStore) processGetResp(mc *MetricsCache, metric *MetricState, resp
 	cs.chunks[0].appender = app
 	cs.chunks[0].state |= chunkStateFirst
 
-	if resp.Error != nil {
-		// assume the item not found   TODO: check error code
-
-		if metric.newName {
-			path := mc.cfg.TablePath + "/names/" + metric.name
-			putInput := v3io.PutItemInput{Path: path, Attributes: map[string]interface{}{}}
-
-			request, err := mc.container.PutItem(&putInput, metric, mc.nameUpdateChan)
-			if err != nil {
-				// Count errors
-				cs.performanceReporter.IncrementCounter("PutNameError", 1)
+	latencyNano := time.Now().UnixNano() - resp.Request().SendTimeNanoseconds
+	cs.performanceReporter.UpdateHistogram("UpdateMetricLatencyHistogram", latencyNano)
+	methodLogger := mc.logger.GetChild("processGetResp")
 
-				mc.logger.ErrorWith("Update-name PutItem failed", "metric", metric.key, "err", err)
-			} else {
-				mc.logger.DebugWith("Update name", "name", metric.name, "key", metric.key, "reqid", request.ID)
+	if resp.Error != nil {
+		if utils.IsNotExistsError(resp.Error) {
+			if metric.newName {
+				path := mc.cfg.TablePath + "/names/" + metric.name
+				putInput := v3io.PutItemInput{Path: path, Attributes: map[string]interface{}{}}
+
+				request, err := mc.container.PutItem(&putInput, metric, mc.nameUpdateChan)
+				if err != nil {
+					// Count errors
+					cs.performanceReporter.IncrementCounter("PutNameError", 1)
+
+					methodLogger.ErrorWith("Update-name PutItem failed", "metric", metric.key, "err", err)
+				} else {
+					methodLogger.DebugWith("Update name", "name", metric.name, "key", metric.key, "reqid", request.ID)
+				}
 			}
+		} else {
+			methodLogger.Error("Update metric has failed with error: %v", resp.Error)
+			// Count errors
```

Comment:
I think you don't have to add this comment everywhere. It's pretty clear what it does :)